sync/block_listener.go

```diff
@@ -50,11 +54,27 @@ func NewBlockListener(net server.Service, bv BlockValidator, layers mesh.Mesh, t
 		Log:            logger,
 		semaphore:      make(chan struct{}, concurrency),
 		unknownQueue:   make(chan mesh.BlockID, 200), //todo tune buffer size + get buffer from config
-		exit:           make(chan struct{})}
+		exit:           make(chan struct{}),
+		receivedGossipBlocks: net.RegisterProtocol(NewBlock),
+	}
 	bl.RegisterMsgHandler(BLOCK, newBlockRequestHandler(layers, logger))
+
+
 	return &bl
 }
 
+func (bl *BlockListener) ListenToGossipBlocks(){
+	for{
+		select {
```

Comment:
should add the block to orphans, also i would do it in the same select loop as run 
another case 
would be much easier to follow and i really don't think we need the parallelism here